ivc/src/ivc/interpreter.rs

```diff
@@ -913,15 +920,101 @@ where
 {
 }
 
+/// Builds selectors for the IVC circuit.
+pub fn build_selectors<F, const N_COL_TOTAL: usize, const N_CHALS: usize>(
+    domain_size: usize,
+) -> Vec<Vec<F>>
+where
+    F: PrimeField,
+{
+    // 3*N + 6*N+2 + N+1 + 35*N + 5 + 1 + N_CHALS =
+    // 45N + 9 + N_CHALS
+    let mut selectors: Vec<Vec<F>> = vec![vec![F::zero(); domain_size]; N_BLOCKS];
+    let mut curr_row = 0;
+    for _i in 0..3 * N_COL_TOTAL {
+        selectors[0][curr_row] = F::one();
+        curr_row += 1;
+    }
+    for _i in 0..6 * N_COL_TOTAL + 2 {
+        selectors[1][curr_row] = F::one();
+        curr_row += 1;
+    }
+    for _i in 0..N_COL_TOTAL + 1 {
+        selectors[2][curr_row] = F::one();
+        curr_row += 1;
+    }
+    for _i in 0..(35 * N_COL_TOTAL + 5) {
+        selectors[3][curr_row] = F::one();
+        curr_row += 1;
+    }
+    for _i in 0..N_CHALS {
+        selectors[4][curr_row] = F::one();
+        curr_row += 1;
+    }
+    for _i in 0..1 {
+        selectors[5][curr_row] = F::one();
+        curr_row += 1;
+    }
+
+    selectors
+}
+
+/// This function generates constraitns for the whole IVC circuit.
+pub fn constrain_selectors<F, Env>(env: &mut Env)
+where
+    F: PrimeField,
+    Env: ColAccessCap<F, IVCColumn>,
+{
+    for i in 0..N_BLOCKS {
```

Comment:
If selectors are public values, i.e. part of the setup, I think we do not need to add constraints.

Reply:
Maybe, but I'm not entirely convinced -- it seems to me that when we're folding the circuit we'll have to constrain these values. For just regular proofs -- yes, it makes sense. I suggest we keep it just in case (it's cheap), and can remove it later. I added a comment above this function. OK with you?

Reply:
Ok to merge like this, but not needed. I would merge the PR, and we can move from there.